graph_weather/models/atmorep/model/atmorep.py

```diff
@@ -0,0 +1,282 @@
+        """
+        return nn.Sequential(
+            nn.Conv2d(1, self.encoder_dim // 2, kernel_size=3, padding=1),
+            nn.ReLU(),
+            nn.BatchNorm2d(self.encoder_dim // 2),
+            nn.Conv2d(self.encoder_dim // 2, self.encoder_dim, kernel_size=3, padding=1),
+            nn.ReLU(),
+            nn.BatchNorm2d(self.encoder_dim),
+        )
+
+    def _create_field_decoder(self):
+        """
+        Creates a CNN decoder for a specific field. This decoder applies convolutional layers
+        followed by ReLU activations and Batch Normalization, and reconstructs the output.
+
+        Returns:
+            nn.Sequential: A CNN decoder for the field.
+        """
+        return nn.Sequential(
+            nn.Conv2d(self.encoder_dim * 2, self.decoder_dim, kernel_size=3, padding=1),
+            nn.ReLU(),
+            nn.BatchNorm2d(self.decoder_dim),
+            nn.Conv2d(self.decoder_dim, self.decoder_dim // 2, kernel_size=3, padding=1),
+            nn.ReLU(),
+            nn.BatchNorm2d(self.decoder_dim // 2),
+            nn.Conv2d(self.decoder_dim // 2, 1, kernel_size=3, padding=1),
+        )
+
+    def forward(self, x, masks=None, **kwargs):
+        """
+        Forward pass through the model. The model encodes field data, applies temporal encoding,
+        and decodes it to make predictions. It also handles optional masking and dropout.
+
+        Args:
+            x (dict): A dictionary containing field-specific input tensors.
+            masks (dict, optional): A dictionary containing field-specific masks for masking predictions.
+            **kwargs: Additional arguments, including ensemble_size for ensemble predictions.
+
+        Returns:
+            dict: A dictionary of predictions for each field.
+        """
+        batch_size = next(iter(x.values())).shape[0]
+        time_steps = next(iter(x.values())).shape[1]
+        height, width = next(iter(x.values())).shape[2:]
+
+        # === Encode each field ===
+        encoded_fields = {}
+        for field, field_data in x.items():
+            B, T, H, W = field_data.shape
+            # Flatten time dimension for 2D CNN processing
+            field_data_flat = field_data.reshape(-1, 1, H, W)
+            encoded = self.field_encoders[field](field_data_flat)
+            # Restore time dimension: (B, T, C, H, W)
+            encoded = encoded.view(batch_size, time_steps, -1, H, W)
+            encoded_fields[field] = encoded
+
+        # === Combine fields and apply temporal encoding ===
+        # Stack over fields: (B, num_fields, T, C, H, W)
+        combined_encoding = torch.stack(
+            [encoded_fields[field] for field in self.input_fields], dim=1
+        )
+        # Average over fields: (B, T, C, H, W)
+        combined_encoding = torch.mean(combined_encoding, dim=1)
+        # Rearrange for 3D conv: (B, C, T, H, W)
+        combined_encoding = combined_encoding.permute(0, 2, 1, 3, 4)
```

Comment:
Can you use `einops` for this? It makes it more readable than torch indicies, same for the other `permute` calls.